### ISSUE TITLE
Unnecessary type dependencies

### ISSUE THREAD

**Issue by bhowell2-fleetio** [pusher/pusher-js#674]
I believe some types were added by mistake to `dependencies` instead of `dev dependencies`. This commit (https://github.com/pusher/pusher-js/commit/ebc3aa7606ad86f3675bc9ab627f50e9a7ce4107) added `@types/express-serve-static-core` as a DEPENDENCY, not a DEV dependency. `@types/express-serve-static-core` also includes other types as deps which may break other people's usage:
```
"@types/express-serve-static-core@4.17.28":
  version "4.17.28"
  resolved "https://registry.yarnpkg.com/@types/express-serve-static-core/-/express-serve-static-core-4.17.28.tgz#c47def9f34ec81dc6328d0b1b5303d1ec98d86b8"
  integrity sha512-P1BJAEAW3E2DJUlkgq4tOL3RyMunoWXqbSCygWo5ZIWTjUgN1YnaXWW4VWl/oc8vs/XoYibEGBKP0uZyF4AHig==
  dependencies:
    "@types/node" "*"
    "@types/qs" "*"
    "@types/range-parser" "*"
```

These shouldn't be exposed this way to consumers of this library.

You also have `@types/node` as a dep and not dev dep as well and I think both should be moved to dev deps.

**Comment by bhowell2-fleetio** [comment]
Here's a PR for it: https://github.com/pusher/pusher-js/pull/675

**Comment by benjamin-tang-pusher** [comment]
Hi, I will ask this team if moving it the @types to devDependencies is necessary.

**Comment by bhowell2-fleetio** [comment]
@benjamin-tang-pusher any word on this? This should be a simple change and there's a PR (#675) for it. Not sure what was up with the failing build on the PR but should have been unrelated as I saw many failing on that time. Maybe a rebase will fix it?

As far as the necessity of this, it is necessary, since these types should not be required for the library to work and they are simply related to the dev environment. If you need to expose any types you do so with `d.ts` that you have generated from your build of this library.

**Comment by fbenevides** [comment]
Released 8.1.0
